### ISSUE TITLE
[Feature Request]: A way to add a metadata to RenderedChunk

### ISSUE THREAD

**Issue by sapphi-red** [rolldown/rolldown#2369]
### What problem does this feature solve?

Vite adds `viteMetadata` property to `RenderedChunk`.
https://github.com/vitejs/vite/blob/7052c8f6fc253f0a88ff04a4c18c108f3bfdaa78/packages/vite/src/node/plugins/metadata.ts#L10-L16
https://github.com/vitejs/vite/blob/7052c8f6fc253f0a88ff04a4c18c108f3bfdaa78/packages/vite/src/node/plugins/asset.ts#L91

Current rolldown/vite workarounds this by having a `Map<string /* chunkName */, ViteMetadata>` for internal usecases.
But Vite needs to expose this information publicly as it's used from many projects ([github search](https://github.com/search?q=%2F%5C.viteMetadata%2F+language%3ATypeScript++NOT+is%3Afork&type=code)).
Vite can expose a function to get this information.

But having a `meta` like the modules have sounds better to me.
https://rollupjs.org/plugin-development/#custom-module-meta-data
In this way, it will probably give a way to access the information from the native side as well.

### What does the proposed API look like?

```js
function accessViteMetadataPlugin() {
	return {
		name: 'accessViteMetadata',
		renderChunk(code, chunk) {
			console.log(chunk.meta.viteMetadata)
			return { code, meta: { viteMetadata: { foo: "bar" } } };
		}
	};
}
```

**Comment by hyf0** [comment]
Yeah. We discuss this before. It's kind of hard to do this. The workaround would be not use the `chunk` `object` itself to stored the related data. You could use standalone place to store the metadata.

There is also other way, which is making rolldown remember the `chunk` `object`. However, we aren't sure if this is the best practice to do such thing. Need more investigations.

**Comment by sapphi-red** [comment]
**_Update:_** I was able to workaround this on vite side.

**Comment by hyf0** [comment]
@sapphi-red After looking into the code, I found we already have a similar mechanism

<img width="1098" height="330" alt="Image" src="https://github.com/user-attachments/assets/2515179b-c11a-43c6-9da4-14423ef3a5ae" />

So I guess we could support this withtout any trouble? Of course, the data will stay in js side.

**Comment by sapphi-red** [comment]
`RenderedChunkMeta` contains an information for the `renderChunk` hook. It doesn't make sense to have that for `generateBundle` & `writeBundle` hooks.

**Comment by hyf0** [comment]
> `RenderedChunkMeta` contains an information for the `renderChunk` hook. It doesn't make sense to have that for `generateBundle` & `writeBundle` hooks.

I wasn't saying reusing/using `RenderChunkMeta` in all places. I refer to it as a proof that we have a mechism to hold the information along the build, which makes no blocker for us to support `chunk.meta`
